### PR TITLE
Handle senses with inflections

### PR DESCRIPTION
**Pull request by craigp-atw**
Before:
image

After:
image

### CODE REVIEW

**Comment by craigp-atw** [comment]
I just realized the Analyzer doesn't even handle senses properly... It basically ignores them (like a note). Should we mimic that behaviour?

File: app/src/routes/lookup/+server.js [hunk 1]
```diff
@@ -61,6 +61,11 @@ function check_inflections(db) {
 	 * @returns {Promise<Stem>}
 	 */
 	async function lookup(possible_inflection) {
+		// strip sense from word
+		let sense_match = possible_inflection.match(/^(.+)(-[A-Z])$/)
```
Comment (longrunningprocess) [inline app/src/routes/lookup/+server.js:65]:
```suggestion
		const sense_match = possible_inflection.match(/^(.+)(-[A-Z])$/)
```

File: app/src/routes/lookup/+server.js [hunk 2]
```diff
@@ -61,6 +61,11 @@ function check_inflections(db) {
...
+		let word = sense_match?.[1] ?? possible_inflection
```
Comment (longrunningprocess) [inline app/src/routes/lookup/+server.js:66]:
```suggestion
		const word = sense_match?.[1] ?? possible_inflection
```

File: app/src/routes/lookup/+server.js [hunk 3]
```diff
@@ -61,6 +61,11 @@ function check_inflections(db) {
...
+		let sense = sense_match?.[2] ?? ''
```
Comment (longrunningprocess) [inline app/src/routes/lookup/+server.js:67]:
```suggestion
		const sense = sense_match?.[2] ?? ''
```

File: app/src/routes/lookup/+server.js [hunk 4]
```diff
@@ -69,6 +74,7 @@ function check_inflections(db) {
 		`
 
 		// prettier-ignore
-		return await db.prepare(sql).bind(`%|${possible_inflection}|%`).first('stem') ?? ''
+		let stem = await db.prepare(sql).bind(`%|${word}|%`).first('stem') ?? ''
```
Comment (longrunningprocess) [inline app/src/routes/lookup/+server.js:77]:
```suggestion
		let stem = await db.prepare(sql).bind(`%|${word}|%`).first('stem') ?? ''
		
```

**Comment by longrunningprocess** [comment]
> I just realized the Analyzer doesn't even handle senses properly... It basically ignores them (like a note). Should we mimic that behaviour?

I don't think so because at some point I want to offer a drop down to the user if they do not put a sense and we find there are senses to a word so they can choose.

**Review by longrunningprocess — APPROVED** [review]
nice!

**Review by longrunningprocess — APPROVED** [review]

**Pull request merged**